### PR TITLE
COL-884, gulp eslint on node_modules/col-* plus fixes

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/COL-884

Notes:
* [`.eslintignore` documentation](http://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories)
* We leverage a double-negative: Ignore default whereby `node_modules` is ignored.
* Too many violations in `col-activities`, etc. to include fixes in this PR.
* Obsolete scripts are ignored
* These are strictly style fixes.

### CODE REVIEW

File: scripts/update_lti.js [hunk 1]
```diff
@@ -106,32 +108,32 @@ var updateToolsForCourse = function(course, index, callback) {
...
-      log.error({'err': err, 'course': course.id}, 'Failed to get external tool configurations for course');
+  CanvasAPI.getExternalToolsForCourse(course, function(apiErr, courseTools) {
+    if (apiErr) {
+      log.error({'err': apiErr, 'course': course.id}, 'Failed to get external tool configurations for course');
       results.error = results.error.concat(toolsToUpdate);
       return callback();
     }
 
-    async.eachOfSeries(toolsToUpdate, function(tool, index, done) {
+    async.eachOfSeries(toolsToUpdate, function(tool, toolIndex, done) {
```
Comment (johncrossman) [inline scripts/update_lti.js]: @paulkerschen I was careful with this renaming of `index`. Please give it a close look, too.

Comment (paulkerschen) [inline scripts/update_lti.js]: It looks right to me as well, @johncrossman. Only thing I'd suggest is that if we rename here, we also rename `index` to `toolIndex` in the sister function `updateToolsForAccount` below; just for consistency.

Comment (johncrossman) [inline scripts/update_lti.js]: Can't argue with consistency, @paulkerschen. Amendment is in.

**Comment by paulkerschen** [comment]
`* @return {Lint}`!

**Pull request merged**